lib/rucio/core/rule.py

```diff
@@ -2707,8 +3199,13 @@ def __evaluate_did_detach(eval_did: models.DataIdentifier, *, session: "Session"
                 files[(file['scope'], file['name'])] = True
             logger(logging.DEBUG, "Removing locks for rule %s [%d/%d/%d]", str(rule.id), rule.locks_ok_cnt, rule.locks_replicating_cnt, rule.locks_stuck_cnt)
             rule_locks_ok_cnt_before = rule.locks_ok_cnt
-            query = session.query(models.ReplicaLock).filter_by(rule_id=rule.id)
-            for lock in query:
+            stmt = select(
+                models.ReplicaLock
+            ).where(
+                models.ReplicaLock.rule_id == rule.id
+            )
+            locks = session.execute(stmt).scalars().all()
+            for lock in locks:
```

Comment:
Similar to my other comments, the following would have been more obvious for me:
```python
for lock in session.execute(stmt).scalars():
```

Reply:
Refactored to:
`for lock in session.execute(stmt).scalars().all():`